ui/app/components/app/signature-request/signature-request-message/signature-request-message.component.js

```diff
@@ -0,0 +1,54 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+
+export default class SignatureRequestMessage extends PureComponent {
+  static propTypes = {
+    data: PropTypes.object.isRequired,
+  }
+
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  makeNode (node) {
+    const leaves = []
+
+    for (const [key, value] of Object.entries(node)) {
+      leaves.push([ key, value ])
+    }
+
+    return leaves
+  }
+
+  renderNode (data) {
+    return (
+      <div className="signature-request-message--node">
+        {this.makeNode(data).map(([ label, value ], i) => (
+          <div className="signature-request-message--node" key={i}>
+            <span className="signature-request-message--node-label">{label}: </span>
+            {
+              typeof value === 'object' && value !== null ?
+                this.renderNode(value)
```

Comment:
ah, nice use of recursion

Reply:
Will arrays show the array index as the "label"? If so, they should probably be handled differently, and just presented as a list of items

Reply:
Is this resolved?

Reply:
Existing designs actually include the array index as the label. So I think my comment here should be ignored. There is perhaps an improvement that could be made, but designs don't account for it and for now I don't want to break expectations. So I am okay with ignoring this if you are are

Reply:
Sounds good